web/src/test/java/se/inera/intyg/webcert/web/service/arende/ArendeDraftServiceImplTest.java

```diff
@@ -90,6 +90,51 @@ public void testSaveDraftUpdate() {
         verifyNoMoreInteractions(repo);
     }
 
+    @Test
+    public void testCreateNew() {
```

Comment:
A newly created ArendeDraft will receive an id from the Repository. Maybe include that in the test.

Reply:
Is it possible to test this when the return from the repo is being mocked?

Reply:
Sure you can. You kind of do this in your new test "shouldReturnSavedArendeDraftFromCreate". The issue there is maybe that it actually just test the scenario when a draft already exists.

You have two behaviours in create(...):
1) If no ArendeDraft exists you create a new one (the new one will receive an id from the db)
2) If a ArendeDraft already exists you should update that one.

Your testCreateNew doesn't really verify the return value. It checks that the repository has been called with the correct arguments. 

It takes a little time to understand the tests because it is not completely clear what they test because they do multiple verification. That is OK, and most pragmatic when it comes to legacy test classes, but I personally find it easier to split into multiple tests with one/fewer asserts per test. 

The test class is quite small so changing it to junit 5 and use nested to do some common mock setup would be quite easy.

You could either do that or you add additional assert(s) to testCreateNew to verify that behaviour #1 also returns the newly created ArendeDraft.

Reply:
I've split up the tests like you suggested, but I still don't understand how I should try that the database sets and id since I have to mock the return from repository.save()

Reply:
Pushed a modification to the test to include the id.